### PR TITLE
[9.0][ADD][website_portal_crm_claim]

### PR DESCRIPTION
**Pull request by yajo**
This module extends the functionality of CRM claims to support showing them in
your portal users' website account page, and to allow them to send messages
from there or create new claims by sending an email to an automatically created
alias.

@Tecnativa

### CODE REVIEW

File: website_portal_crm_claim/README.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+   a new claim, and the list of claims where you are subscribed.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/186/9.0
+
+Known issues / Roadmap
+======================
+
+``website_crm_claim`` addon is a very ugly addon that just adds a link to
```
Comment (pedrobaeza) [inline website_portal_crm_claim/README.rst]: Add bullets for each point `*`

File: website_portal_crm_claim/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "version": "9.0.1.0.0",
+    "category": "Website",
+    "website": "https://www.tecnativa.com/",
+    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "license": "LGPL-3",
+    "application": False,
+    "installable": True,
+    "depends": [
+        "crm_claim",
+        "website_crm_claim",  # TODO See known issues; drop in v10
```
Comment (pedrobaeza) [inline website_portal_crm_claim/__openerp__.py]: Better use FIXME key for this

File: website_portal_crm_claim/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        "website_portal_v10",
```
Comment (pedrobaeza) [inline website_portal_crm_claim/__openerp__.py]: Do we need to depend on this module?

File: website_portal_crm_claim/wizards/sale_config_settings.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+# Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from openerp import api, fields, models
+
+
+class SaleConfigSettings(models.TransientModel):
+    _inherit = "sale.config.settings"
+
+    alias_claim = fields.Char(
```
Comment (pedrobaeza) [inline website_portal_crm_claim/wizards/sale_config_settings.py:11]: Why not use directly alias widget and call this `claim_alias_id`?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: website_portal_crm_claim/__openerp__.py (hunk 3 above)
Comment (yajo) [reply website_portal_crm_claim/__openerp__.py]: Absolutely.
https://github.com/OCA/website/pull/266/files#diff-751a21571f0e7db22d65419c12df7c38R8

File: website_portal_crm_claim/wizards/sale_config_settings.py (hunk 4 above)
Comment (yajo) [reply website_portal_crm_claim/wizards/sale_config_settings.py:11]: I'm trying to imitate here what Odoo core does.
https://github.com/odoo/odoo/blob/fc2e80cb4bcc450762c7ac5cb82a3e2d88062b38/addons/crm/models/res_config.py#L14
https://github.com/odoo/odoo/blob/fc2e80cb4bcc450762c7ac5cb82a3e2d88062b38/addons/crm/models/res_config.py#L48-L62

Comment (pedrobaeza) [reply website_portal_crm_claim/wizards/sale_config_settings.py:11]: OK

File: website_portal_crm_claim/README.rst (hunk 1 above)
Comment (pedrobaeza) [reply website_portal_crm_claim/README.rst]: This point is still missing

**Review by pedrobaeza — APPROVED** [review]
Squashing on merging

**Comment by rafaelbn** [comment]
Hi @Yajo , travis fails, could you check it please? Thanks

**Review by rafaelbn — APPROVED** [review]
tested in runbot

**Comment by pedrobaeza** [comment]
Why is Travis red? Maybe a rebase?

**Comment by yajo** [comment]
Most likely it was https://github.com/OCA/website/pull/346. Rebased, let's see.

File: website_portal_crm_claim/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+           type="http", auth="user", website=True)
+    def portal_my_claims(self, page=1, date_begin=None, date_end=None,
+                         **kwargs):
+        values = self._prepare_portal_layout_values()
+        Claim = request.env["crm.claim"]
+        url = "/my/claims"
+        mailto = False
+        alias = (request.env["sale.config.settings"]
+                 ._find_default_claim_alias_id())
+        if "@" in alias.display_name and alias.alias_domain != "localhost":
```
Comment (andriusdaunoravicius) [inline website_portal_crm_claim/controllers/main.py:42]: Weak email validation, use [validate_email](https://pypi.python.org/pypi/validate_email) library, or at least use a regex.

**Comment by pedrobaeza** [comment]
Closing as the project is no longer active.

**Pull request closed without merging**